core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2449,7 +2459,12 @@ private[spark] class DAGScheduler(
     val currentEpoch = maybeEpoch.getOrElse(mapOutputTracker.getEpoch)
     logDebug(s"Considering removal of executor $execId; " +
       s"fileLost: $fileLost, currentEpoch: $currentEpoch")
-    if (!executorFailureEpoch.contains(execId) || executorFailureEpoch(execId) < currentEpoch) {
+    // Check if the execId is a shuffle push merger
+    // We do not remove the executor if it is,
+    // and only remove the outputs on that host/executor.
```

Comment:
Nit: `and only remove the outputs on the host`

Reply:
done